### PR TITLE
[WIP] Modularized code

### PR DESCRIPTION
**Pull request by nickbalestra**
@lmatteis in direction of adding tests, I've modularised the code as it was incapsulating everything, this way should be easier to unit test it as we now have separate modules for:

- actionDriver
- stateDriver
- combineCycles
- createCycleMiddleware

### CODE REVIEW

File: src/createCycleMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import {run} from '@cycle/xstream-run';
+import actionDriver, { actionListener } from './drivers/actionDriver';
+import stateDriver, { stateListener } from './drivers/stateDriver';
```
Comment (goshakkk) [inline src/createCycleMiddleware.js]: Honestly, this looks kinda odd and singleton-ish. Maybe we should still init these in `createCycleMiddleware` *and* pass them to the driver, along with `store`?

Comment (goshakkk) [inline src/createCycleMiddleware.js]: Maybe have the drivers do this:

```javascript
const action = actionDriver(store);
const { actionListener } = action;
drivers.ACTION = action.driver;
```

or something.

Or maybe still leave both of these drivers inlined, it wasn't super bad.

Comment (nickbalestra) [inline src/createCycleMiddleware.js]: I'm also fiddling with it... but either something like you describe, ie

```javascript
const { actionDriver, actionListener } = createActionDriver(store)
```

or leaving both drivers inlined. 
What are your thoughts? Especially in relation to testing it, that will be next

**Comment by nickbalestra** [comment]
I think that without relying on the 'singleton-ish' part will be hard to make this work...(I've tried to implement @goshakkk idea without luck). If we believe that we don't need to add unit test, you can close this, otherwise I don't see many alternatives.

**Comment by goshakkk** [comment]
Then maybe let's just extract the cycle merge utility into a separate file, but keep the drivers inline?

**Comment by nickbalestra** [comment]
@I'll add some unit tests next, then this should be also ready to go.

**Comment by lmatteis** [comment]
Ok I've enabled travis. I'm not sure why it's not showing up for existing PRs. Maybe once you push?

**Comment by nickbalestra** [comment]
Rebased and push, all working as expected :) thanks!

**Pull request merged**